lib/session.js

```diff
@@ -92,6 +92,7 @@ var doSignin = function(lookup, req, res, onSuccess, onFail) {
 					if (!err && isMatch) {
 						postHookedSigninWithUser(user, req, res, onSuccess, onFail);
 					} else {
+						err = new Error('The email and password you entered do not match.');
```

Comment:
You are overriding here the original error. This may not always be a wise choice.